frontend/src/tests/posting-tests.js

```diff
@@ -1,39 +1,354 @@
+    it("Modify a posting_position", async () => {
+        const updatePostingPos = {
+            position_id: postingPos.position_id,
+            hours: 10,
+            num_positions: 2,
+        };
+
+        resp = await apiPOST(
+            `/admin/postings/${posting.id}/posting_positions`,
+            updatePostingPos
+        );
+        expect(resp).toHaveStatus("success");
+        expect(resp.payload).toMatchObject(updatePostingPos);
+    });
+
+    it("Cannot create two posting_positions with the same position_id for a single posting", async () => {
+        const newPostingPos = {
+            position_id: position.id,
+            hours: 10,
+            num_positions: 2,
+        };
+
+        resp = await apiPOST(
+            `/admin/postings/${posting.id}/posting_positions`,
+            newPostingPos
+        );
+        expect(resp).toHaveStatus("success");
+        resp = await apiGET(`/admin/postings/${posting.id}/posting_positions`);
+
+        // The previous POST request should not create a new posting_position and instead it upserts to an existing one
+        expect(
+            resp.payload.filter((p) => p.position_id === position.id).length
+        ).toEqual(1);
+    });
+    it("Can create two posting_positions with the same position_id for different postings", async () => {
+        const newPosting = {
+            name: "CSC500F TA",
+            intro_text: "Posting for CSC500F",
+            open_date: "2021/04/01",
+            close_date: "2021/05/01",
+            availability: "auto",
+        };
+
+        resp = await apiPOST(
+            `/admin/sessions/${session.id}/postings`,
+            newPosting
+        );
+        expect(resp).toHaveStatus("success");
+        Object.assign(newPosting, resp.payload);
+
+        const newPostingPos = {
+            position_id: position.id,
+            hours: 10,
+            num_positions: 2,
+        };
+
+        resp = await apiPOST(
+            `/admin/postings/${newPosting.id}/posting_positions`,
+            newPostingPos
+        );
+
+        expect(resp).toHaveStatus("success");
+        resp = await apiGET(
+            `/admin/postings/${newPosting.id}/posting_positions`
+        );
+        // Verify a new posting_position is created successfully for the new posting
+        expect(
+            resp.payload.filter((p) => p.position_id === position.id).length
+        ).toEqual(1);
+    });
+
+    it("Delete a posting_position", async () => {
+        resp = await apiPOST(`/admin/posting_positions/delete`, {
+            posting_id: posting.id,
+            position_id: position.id,
+        });
+        expect(resp).toHaveStatus("success");
+        resp = await apiGET(`/admin/postings/${posting.id}/posting_positions`);
+        expect(
+            resp.payload.filter((p) => p.position_id === position.id).length
+        ).toEqual(0);
+    });
+
+    it.skip("Deleting a posting also deletes all associated posting_positions", async () => {
```

Comment:
why is this test with `skip`

Reply:
Because there is an issue need to be fixed at the backend. Skipped for now.